### PR TITLE
use diff intermediate var to preserve functionality

### LINKED ISSUE #83737: Ansible throws an error about "raw parameters" in shell task
### Summary

Commit https://github.com/ansible/ansible/commit/885e3766a8d637c00254d25ea91e77733cc904b8 broke a shell module run.
Run the task:
```yaml
- name: Add string to change the systemd unit file
  ansible.builtin.shell: echo 'test=onetwo' >> /tmp/some.service
```
It fails with error:
```
fatal: [localhost]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python3.12"
    },
    "changed": false
}

MSG:

Unsupported parameters for (ansible.legacy.command) module: 'test. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends.
```

### Issue Type

Bug Report

### Component Name

shell

### Ansible Version

```console
$ ansible --version
ansible-core-2.18.0.dev0
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
```

### OS / Environment

All

### Steps to Reproduce

```yaml (paste below)
- ansible.builtin.shell: echo 'test=onetwo' >> /tmp/some.service
```

### Expected Results

Before this commit it adds string "test=onetwo" to the file without errors.

### Actual Results

```console
fatal: [localhost]: FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python3.12"
    },
    "changed": false
}

MSG:

Unsupported parameters for (ansible.legacy.command) module: 'test. Supported parameters include: _raw_params, _uses_shell, argv, chdir, creates, executable, expand_argument_vars, removes, stdin, stdin_add_newline, strip_empty_ends.
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by bcoca**
fixes #83737
add tests 

##### ISSUE TYPE

- Bugfix Pull Request

### CODE REVIEW

File: test/integration/targets/run_modules/runme.sh [hunk 1]
```diff
@@ -4,3 +4,11 @@ set -eux
...
 # test running module directly
 python.py library/test.py args.json
+
+TMPFILE=$(shell mktemp -p "${OUTPUT_DIR}" 2>/dev/null || mktemp -t 'ansible-testing-XXXXXXXXXX' -p "${OUTPUT_DIR}")
+
+# ensure 'command' can use 'raw args'
+ansible -m command -a "dd if=/dev/zero of=\"${TMPFILE}\" bs=1K count=1" localhost
+
+# ensure fqcn 'command' can use 'raw args'
+ansible -m ansible.legacy.command -a "dd if=/dev/zero of=\"${TMPFILE}\" bs=1K count=1" localhost
```
Comment (s-hertel) [inline test/integration/targets/run_modules/runme.sh:14]: I can only reproduce the bug with a task in a file. The adhoc task ds ends up injecting args as `_raw_params` which avoids the issue.

**Review by s-hertel — APPROVED** [review]
Sorry for overlooking this, the fix looks good to me.

**Comment by sshnaidm** [comment]
Tested with 
```yaml
- ansible.builtin.shell: echo 'test=' >> /tmp/tempfile
```
and the issue is fixed.

**Pull request merged**